core/controllers/feature_flag_test.py

```diff
@@ -0,0 +1,147 @@
+            self.prod_feature_flag.feature_flag_value.name, True, 0, []
+        )
+
+        # Here we use MyPy ignore because the expected type of ALL_FEATURE_FLAGS
+        # is a list of 'platform_feature_list.FeatureNames' Enum, but here for
+        # testing purposes we are providing a list of custom 'FeatureNames'
+        # enums for mocking the actual behavior, which causes MyPy to throw an
+        # 'Incompatible types in assignment' error. Thus to avoid the error, we
+        # used ignore here.
+        feature_services.ALL_FEATURE_FLAGS = feature_name_enums  # type: ignore[assignment]
+        feature_services.ALL_FEATURES_NAMES_SET = set(feature_names)
+
+    def tearDown(self) -> None:
+        super().tearDown()
+
+        feature_services.ALL_FEATURE_FLAGS = self.original_feature_list
+        feature_services.ALL_FEATURES_NAMES_SET = self.original_feature_name_set
+        registry.Registry.feature_flag_spec_registry = self.original_registry
+
+    def test_feature_flag_evaluation_is_correct(self) -> None:
+        result = self.get_json(
+            '/feature_flags_evaluation_handler'
+        )
+        self.assertEqual(
+            result,
+            {
+                self.dev_feature_flag.feature_flag_value.name: False,
+                self.prod_feature_flag.feature_flag_value.name: True
+            }
+        )
+
+
+class FeatureFlagDummyHandlerTest(test_utils.GenericTestBase):
+    """Tests for the FeatureFlagDummyHandler."""
+
+    def tearDown(self) -> None:
+        feature_services.update_feature_flag(
+            platform_feature_list.FeatureNames.
+            DUMMY_FEATURE_FLAG_FOR_E2E_TESTS.value,
+            False,
+            0,
+            []
+        )
+
+        super().tearDown()
+
+    def _set_dummy_feature_flag_status(
+        self, feature_is_enabled: bool) -> None:
+        """Sets the dummy_feature feature flag value."""
+        feature_services.update_feature_flag(
+            platform_feature_list.FeatureNames.
+            DUMMY_FEATURE_FLAG_FOR_E2E_TESTS.value,
+            feature_is_enabled,
+            0,
+            []
+        )
+
+    def test_get_with_dummy_feature_flag_enabled_returns_ok(self) -> None:
+        self.get_json(
+            '/feature_flag_dummy_handler',
+            expected_status_int=404
+        )
+
+        self._set_dummy_feature_flag_status(True)
+        result = self.get_json(
+            '/feature_flag_dummy_handler',
+        )
+        self.assertEqual(result, {'msg': 'ok'})
+
+    def test_get_with_dummy_feature_flag_disabled_raises_404(self) -> None:
+        self.get_json(
+            '/feature_flag_dummy_handler',
+            expected_status_int=404
+        )
+        self._set_dummy_feature_flag_status(True)
+        self.get_json(
+            '/feature_flag_dummy_handler',
+            expected_status_int=200
+        )
+
+        self._set_dummy_feature_flag_status(False)
+        self.get_json(
```

Comment:
Ah right, Actually, we have a separate handler for the dummy feature flag, and it's been there for a while. I think the reason is that we only want to make the request for the dummy feature-flag when needed as this is only for the testing purpose, all the other feature flags are called together as we tend to get feature flag evaluation values more often.
I have updated the handler to return the proper response to the frontend, instead of 404 we are actually returning the value, whether the dummy feature flag is on or not.